client/verta/tests/test_artifacts.py

```diff
@@ -88,6 +91,42 @@ def test_upload_dir(self, experiment_run, strs, dir_and_files):
         with zipfile.ZipFile(experiment_run.get_artifact(key), 'r') as zipf:
             assert filepaths == set(zipf.namelist())
 
+    @pytest.mark.not_oss
+    # NOTE: `capsys` prevents prints within this test from being output to logs
+    def test_upload_multipart(self, capsys, experiment_run, strs):
+        key = strs[0]
+
+        # for verifying that multipart upload occurs
+        get_parts_url = "{}://{}/api/v1/modeldb/experiment-run/getCommittedArtifactParts".format(
+            experiment_run._conn.scheme,
+            experiment_run._conn.socket,
+        )
+        get_parts_params = {
+            'id': experiment_run.id,
+            'key': key,
+        }
+
+        FILE_SIZE = 6*10**6  # 6 MB
+        with tempfile.NamedTemporaryFile(suffix='.txt') as tempf:
+            tempf.truncate(FILE_SIZE)  # zero-filled
+            tempf.flush()  # flush object buffer
+            os.fsync(tempf.fileno())  # flush OS buffer
+
+            # no committed artifact parts yet
+            response = _utils.make_request("GET", get_parts_url, experiment_run._conn, params=get_parts_params)
+            assert response.status_code == 404
+            assert "Can't find specified artifact" in response.text
+
+            # log artifact
+            experiment_run.log_artifact(key, tempf)
+
+            # Client entered multipart loop
+            assert "uploading part " in capsys.readouterr().out
+
+            # artifact parts ModelDB entry exists
```

Comment:
Can we make a stronger assertion?

Reply:
I couldn't come up with a stronger assertion to make, since the committed artifact parts are deleted from the database once the multipart upload is completed (see [my comment on Evgenii's backend PR](https://github.com/VertaAI/modeldb/pull/660#pullrequestreview-418686850) for context).

Do you suppose this would be a good reason to keep them in the database?

Reply:
Maybe. But if that's the case, I don't know what the call below is doing.

Reply:
It just checks that `getCommittedArtifactParts` now returns a non-error response (it returned a `404` before `log_artifact()`), suggesting (weakly) that MDB registered committed artifact points at some point

It's indirect, but it's the most I could do with the information I seem to have.

Reply:
That's a very weak test that doesn't verify it was actually uploaded with multiple parts. If you're missing information, then we should consider exposing that information.

It's also weird to me that it returning not-404 while also deleting the data.